toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -193,11 +197,37 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 	)
 
 	logBaseName := filepath.Base(srpmFile) + ".log"
+	const checkAttempts = 3
```

Comment:
I still don't like the hard-coding of the attempts count. Here's my current suggestion:

Let's extend the meaning of `PACKAGE_BUILD_RETRIES` (found inside the `Makefile` file) to also force a rebuild in case of a test failure. Then let's change its default from 1 to 3. This should allow us to directly use `buildAttempts` in this function and completely drop `checkAttempts` along with some of the code. We'd also then have to slightly update `building.md` to mention, that the rebuilds will also happen for failed package tests.

I'm aware this is not the perfect solution, but the best, no overly complicated idea that came to my mind, considering we don't currently have a way to re-run only the `%check` section for otherwise successful builds. I'm open to other ideas, I don't want to hard-code these retries, though.

Reply:
Would I change the default from 1 to 3 for all scenarios, or would it be something like default is 3 if RUN_CHECK=y, else 1.

I think that there's a way to run %check without running %build, and that would be using `rpmbuild -bi --short-circuit`, which skips all steps before %install. If we do that, though, that would likely involve restructuring the way we run `rpmbuild`.

Reply:
Yes, ideally I'd only change the default for `RUN_CHECK=y`. Good idea!

And my gut is telling me, that trying to modify how we run `rpmbuild` might be a rabbit hole, so I didn't want to suggest it for this change. Having said that, if you can give it a try and see if it can be done within a few short hours, then go for it! If that won't work, or there's no time for this now, please create a bug summarizing what we did here and what might be a potential improvement, so at least we have a good starting point for future work.

Reply:
Update: discussed offline and my idea wouldn't work well, since it would make the build process somewhat surprising for test runs due to additional rebuilds. Current plan is to modify the rebuilds through setting the `PACKAGE_BUILD_RETRIES` argument during test runs.